internal/adminanalytics/users.go

```diff
@@ -117,151 +137,89 @@ func (n *UsersFrequencyNode) Frequency() float64 { return n.Data.Frequency }
+	prevYear := year
 
-func (s *Users) Summary(ctx context.Context) (*UsersSummary, error) {
-	now := time.Now()
-	from, err := getFromDate(s.DateRange, now)
-	if err != nil {
-		return nil, err
+	if prevMonth == time.December && prev > 0 {
+		prevYear = year - 1
+	}
+
+	if int(prevMonth) < 10 {
+		return fmt.Sprintf("%d-0%d", prevYear, int(prevMonth))
 	}
 
-	cacheKey := fmt.Sprintf("Users:%s:%s", "Summary", s.DateRange)
+	return fmt.Sprintf("%d-%d", prevYear, int(prevMonth))
+}
 
-	if s.Cache == true {
-		if summary, err := getItemFromCache[UsersSummary](cacheKey); err == nil {
-			return summary, nil
+var (
+	mauQuery = `
+	SELECT 
+		TO_CHAR(timestamp, 'YYYY-MM') AS date,
+		COUNT(DISTINCT CASE WHEN user_id = 0 THEN anonymous_user_id ELSE CAST(user_id AS TEXT) END) AS count
+	FROM event_logs
+	WHERE 
+		anonymous_user_id <> 'backend' 
+		AND TO_CHAR(timestamp, 'YYYY-MM') IN (%s)
+		AND name NOT IN (%s)
+	GROUP BY 1
+	ORDER BY 1 ASC
+	`
+)
+
+func (f *Users) Maus(ctx context.Context) ([]*UsersMAUNode, error) {
+	cacheKey := fmt.Sprintf("Users:%s", "MAU")
+	if f.Cache == true {
+		if nodes, err := getArrayFromCache[UsersMAUNode](cacheKey); err == nil {
+			return nodes, nil
 		}
 	}
-	query := sqlf.Sprintf(fmt.Sprintf(avgUsersByPeriodQuery, from.Format("2006-01-02 15:04:05"), now.Format("2006-01-02 15:04:05")))
-	rows, err := s.DB.QueryContext(ctx, query.Query(sqlf.PostgresBindVar), query.Args()...)
 
+	now := time.Now()
+	var months []*sqlf.Query
+	for _, prev := range [3]int{2, 1, 0} {
+		months = append(months, sqlf.Sprintf("%s", getPrevMonthString(now, prev)))
+	}
+	query := sqlf.Sprintf(mauQuery, sqlf.Join(months, ","), sqlf.Join(nonActiveUserEvents, ","))
+
+	rows, err := f.DB.QueryContext(ctx, query.Query(sqlf.PostgresBindVar), query.Args()...)
 	if err != nil {
 		return nil, err
 	}
-
 	defer rows.Close()
 
-	var summaryData UsersSummaryData
-	for i := 0; i < 3; i++ {
-		rows.Next()
+	nodes := make([]*UsersMAUNode, 0, 3)
+	for rows.Next() {
+		var data UsersMAUNodeData
 
-		var totalCount float64
-		var metric string
-		if err := rows.Scan(&metric, &totalCount); err != nil {
+		if err := rows.Scan(&data.Date, &data.Count); err != nil {
 			return nil, err
 		}
 
-		if metric == "DAU" {
-			summaryData.AvgDAU = totalCount
-		} else if metric == "WAU" {
-			summaryData.AvgWAU = totalCount
-		} else if metric == "MAU" {
-			summaryData.AvgMAU = totalCount
-		}
+		nodes = append(nodes, &UsersMAUNode{data})
 	}
 
-	summary := UsersSummary{summaryData}
-
-	if _, err := setItemToCache(cacheKey, &summary); err != nil {
+	if _, err := setArrayToCache(cacheKey, nodes); err != nil {
 		return nil, err
 	}
 
-	return &summary, nil
+	return nodes, nil
 }
 
-type UsersSummaryData struct {
-	AvgDAU float64
-	AvgWAU float64
-	AvgMAU float64
+type UsersMAUNodeData struct {
+	Date  string
+	Count float64
 }
 
-type UsersSummary struct {
-	Data UsersSummaryData
+type UsersMAUNode struct {
```

Comment:
`Node` seems a bit out of place when talking about database rows.

```suggestion
type MonthlyActiveUsersRow struct {
```

Reply:
updated